indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/pki/X509Certificate.kt

```diff
@@ -235,9 +263,10 @@ val CryptoSignature.x509Encoded
  * - EC is DER-encoded then wrapped in a bit string
  */
 fun CryptoSignature.Companion.fromX509Encoded(alg: X509SignatureAlgorithm, it: Asn1Primitive) =
-    when (alg.isEc) {
-        true -> CryptoSignature.EC.decodeFromDer(it.asAsn1BitString().rawBytes)
-        false -> CryptoSignature.RSA.decodeFromTlv(it)
+    when (alg) {
+        is X509SignatureAlgorithm.EC -> CryptoSignature.EC.decodeFromDer(it.asAsn1BitString().rawBytes)
+        is X509SignatureAlgorithm.RSA -> CryptoSignature.RSA.decodeFromTlv(it)
+        else -> throw IllegalArgumentException("Unsupported signature algorithm.")
```

Comment:
maybe leave a comment here so that we don't miss updating this when we make our core data classes equally extensible?